Add more information about how `mike` works (#102)
frequenz-floss/frequenz-repo-config-python: docs/index.md

If all went well, your website should be available soon via
`https://<repo-owner>.github.io/<repo-name>/`.

#### Website versions using Mike

The above commands create a new documentation version using
[Mike](https://pypi.org/project/mike/), which is used to keep multiple versions
of the website.

The new documentation version is called `next`, which is used as the name for
the currently in-development branch. The `next` branch has an alias called
`latest`, which is set as the *default*.

If the website is visited without specifying an explicit version, the `latest`
version will be displayed. It is recommended to point `latest` to the latest

> stable version instead of the currently in-devepoment version, so as you as you

Reviewer: duplicated as you as you

Sorry I've just spotted it

Author: Actually `s/you/soon/`: "as soon as you" :)

Author: Fixed.

make a release, you should update the alias.

#### Auto-generation of pages via GitHub Actions

New versions of the documentation will be automatically generated and published
via GitHub Actions on any push.

If a push is to a branch with semver-like format (vX.Y.Z), then the generated
version will replace the current `next` version.

If a tag is pushed instead, then the generated version will replace the current
`vX.Y` version (if there was any), and the aliases `vX` and `latest` will be
updated to point to the new version.

> So basically, we don't create versions for patch releases, only for minor

Reviewer 2: ```suggestion
To summarize, we don't create versions for patch releases, only for minor
```
